### DISCUSSION TITLE
Set session variable after_create_account

### DISCUSSION THREAD

**Discussion by learn-ai763** [janko/rodauth-rails#325] [General]
Hi I'm wondering how can I set session variable in `after_create_account` hook

```
after_create_account do
  session[:current_company_id]  = 11
  Rails.logger.debug("Session company id: #{session[:current_company_id]}")
end
```

The problem is, when the user signs up and get redirected to the root_url the session variable is not set. Even though the block is definitely evaluated

**Comment by janko** [comment]
If you have autologin enabled on account creation (true if you have `verify_account_grace_period` feature loaded), that will reset the session the logging the user in. This is to prevent session fixation attacks, but it means the session will be cleared.

One way to work around this is to hook onto something that's run after autologin, such as `create_account_response`:

```rb
create_account_response do
  session[:current_company_id] = 11
  super()
end
```

**Reply by learn-ai763** [reply]
Indeed, it works! Thank you, this solved the issue.
